internal/server_test.go

```diff
@@ -103,16 +103,16 @@ func TestServerAuthHandlerValid(t *testing.T) {
 
 	// Should allow valid request email
 	req := newDefaultHttpRequest("/foo")
-	c := MakeCookie(req, "test@example.com")
+	c := MakeCookie(req, generateTestAuthMethod("test@example.com"))
 	config.Domains = []string{}
```

Comment:
It would make sense to also `config.Whitelist = []string{"test@example.com"}` so that that process is checked. Currently if you add that to the test it fails due to error.

Reply:
I think this fails because the newly introduced `authMethod` has character escaped ampersand and the whitelist holds only the un-escaped version. And `ValidateEmail()` method only checks that whitelist verbatim without any character decoding

Reply:
It should be fixed in d4fba8c, thanks for the review 👍